src/diff.c

```diff
@@ -178,7 +178,7 @@ diff_common_add_diff_stat(struct view *view, const char *text, size_t offset)
 
 	/* Ensure that '|' is present and the file name part contains
 	 * non-space characters. */
-	if (!pipe || pipe == data || strcspn(data, " ") == 0)
+	if (!pipe || pipe == data || strlen(pipe) == len)
```

Comment:
I'm trying to wrap my head around this one. Would be good to maybe extend the comment a bit with an example or provide a diffstat where this fixes a false positive.

I will work on making a column renderer which prints the line type. This would allow us to make this testable.